File: src/main/java/de/fluxparticle/animation/AnimationCollection.java
package de.fluxparticle.animation;

import java.util.Collection;
import java.util.LinkedList;
import java.util.SortedMap;
import java.util.TreeMap;

public class AnimationCollection {

	private final SortedMap<Integer, Collection<Timeframe>> animations = new TreeMap<>();

	public void add(int priority, Timeframe Timeframe) {
        animations.computeIfAbsent(priority, k -> new LinkedList<>()).add(Timeframe);
	}

	public void appendToClip(Clip clip) {
		for (Collection<Timeframe> values : animations.values()) {
			clip.invoke(() -> values.forEach(clip::addTimeframe), false);
		}
	}

    public void appendToClipRaw(Clip clip) {
        for (Collection<Timeframe> values : animations.values()) {
            values.forEach(clip::addTimeframe);
        }
    }

}

File: src/main/kotlin/de/fluxparticle/animation/AnimationCollection.kt
package de.fluxparticle.animation

import kotlin.collections.HashMap
import kotlin.collections.ArrayList

class AnimationCollection {

    private val animations: MutableMap<Int, MutableList<Timeframe>> = HashMap()

    fun add(priority: Int, Timeframe: Timeframe) {
        var get = animations.get(priority)
        if (get == null) {
            get = ArrayList()
            animations.put(priority, get!!)
        }
        get.add(Timeframe)
    }

    fun appendToClip(clip: Clip) {
        for (values in animations.values) {
            clip.invoke(false) {
                values.forEach {
                    clip.addTimeframe(it)
                }
            }
        }
    }

    fun appendToClipRaw(clip: Clip) {
        for (values in animations.values) {
            values.forEach { clip.addTimeframe(it) }
        }
    }

}
